test/integration/hds_integration_test.cc

```diff
@@ -145,6 +145,40 @@ class HdsIntegrationTest : public HttpIntegrationTest,
     return server_health_check_specifier_;
   }
 
+  // Creates a basic HealthCheckSpecifier message containing one endpoint and
+  // one TCP health_check
+  envoy::service::discovery::v2::HealthCheckSpecifier makeTCPHealthCheckSpecifier() {
+    envoy::service::discovery::v2::HealthCheckSpecifier server_health_check_specifier_;
+    server_health_check_specifier_.mutable_interval()->set_seconds(1);
+
+    auto* health_check = server_health_check_specifier_.add_health_check();
+
+    health_check->set_cluster_name("anna");
+    health_check->add_endpoints()
+        ->add_endpoints()
+        ->mutable_address()
+        ->mutable_socket_address()
+        ->set_address(host_upstream_->localAddress()->ip()->addressAsString());
+    health_check->mutable_endpoints(0)
+        ->mutable_endpoints(0)
+        ->mutable_address()
+        ->mutable_socket_address()
+        ->set_port_value(host_upstream_->localAddress()->ip()->port());
```

Comment:
Nit: consider using `addressToProtobufAddress()` in `source/common/network/utility.h` for this (and elsewhere in the HDS tests).